srtcore/core.cpp

```diff
@@ -11192,6 +11233,8 @@ void CUDT::checkTimers()
 
 void CUDT::updateBrokenConnection()
 {
+    HLOGC(smlog.Debug, log << "updateBrokenConnection: setting closing=true and taking out epoll events");
+    m_bClosing = true;
```

Comment:
Does this fix require other changes introduced in the PR?

Reply:
This is simply a completion for all cases when `releaseSync` is called, which is connected to the fact that `releaseSync` is about to make TSBPD thread exit, and it can only exit if `m_bClosing == true`.

Reply:
Does only this change make sense without the rest changes in this PR?

Reply:
Rest of the changes are formal fixes for real problems, others - including this one - are "sanity fixes". This flag must be set true when the TSBPD thread is expected to exit otherwise it risks a deadlock. Relying on that a flag is set appropriately before exitting isn't the best approach, but it's simply derived from UDT and it's a consequence of how the TSBPD thread is expected to work. Changes here were due to an in-between lifted lock on `m_RecvLock`, which was newly introduced with the groups.

Reply:
Let's move this fix to a separate PR, so it is not blocked by the review of other changes introduced here.